lib/models/LifecycleConfiguration.js

```diff
@@ -547,6 +547,50 @@ class LifecycleConfiguration {
         return null;
     }
 
+    /**
+     * Ensure that transition rules are at least a day apart from each other.
+     * @param {object} params - Given function parameters
+     * @param {string} [params.days] - The days of the current transition
+     * @param {string} [params.date] - The date of the current transition
+     * @param {string} params.storageClass - The storage class of the current
+     * rule
+     * @param {string} params.rule - The current rule
+     * @return {undefined}
+     */
+    _checkTimeGap(params) {
+        // For testing purposes, we want to allow smaller time intervals.
+        if (process.env.CI === 'true') {
+            return undefined;
+        }
+        const { days, date, storageClass, rule } = params;
+        const invalidTransition = rule.Transition.find(transition => {
+            if (storageClass === transition.StorageClass[0]) {
+                return undefined;
+            }
+            if (days !== undefined) {
+                return Number.parseInt(transition.Days[0], 10) === days;
+            }
+            if (date !== undefined) {
+                const timestamp = new Date(date).getTime();
+                const compareTimestamp = new Date(transition.Date[0]).getTime();
+                const oneDay = 24 * 60 * 60 * 1000; // Milliseconds in a day.
+                return Math.abs(timestamp - compareTimestamp) < oneDay;
+            }
+            return undefined;
+        });
+        if (invalidTransition) {
+            const timeType = days !== undefined ? 'Days' : 'Date';
+            const filterMsg = this._getRuleFilter(rule);
+            const compareStorageClass = invalidTransition.StorageClass[0];
+            const msg = `'${timeType}' in the 'Transition' action for ` +
+                `StorageClass '${storageClass}' for ${filterMsg} must be 1 ` +
```

Comment:
minor suggestion: I would rephrase to "must be different than" or "must be at least one day apart" in place of "must be 1 day more than", as the rule is symmetric to both rules and can be more than one day apart (i.e. one can be one or more days more than the other)